### DISCUSSION TITLE
filter needs to ancestors/descendants

### DISCUSSION THREAD

**Discussion by danieleades** [useblocks/sphinx-needs#617] [Q&A]
I'd like to produce a `needflow` that contains only ancestors or descendants of a given need. Is there an easy way to do this?

To be clear, I want to include *all* ancestors and descendants, not just immediate ones

**Comment by vpoughonEZ** [comment]
Relevant to my other post https://github.com/useblocks/sphinxcontrib-needs/discussions/615. But beware it is quite inefficient.

**Comment by danieleades** [comment]
@danwos  is there a convenient way to at least get the immediate parents or children of a need?

**Comment by danwos** [comment]
I'm afraid not. You only have the IDs in the different link-options and you need to look them up by your own.
So, huge room for improvement...

**Reply by danieleades** [reply]
in the case, what are the 'links' and 'links_back' attributes used for in a filter string?

in my case, I have a number of 'work packages' which each deliver multiple requirements. On the page for each work package I'd like to show a `needflow` which shows all the requirements linking to it. Is that possible?

i've tried something like
```rst
.. needflow::
    :filter: "WP-1" in links_back
```
or

```rst
.. needflow::
    :filter: "WP-1" in links
```

how is this supposed to be used?

**Reply by danieleades** [reply]
never mind, figured it out

**Comment by danieleades** [comment]
i made a proof of concept extension that builds a directed graph of 'nodes' - https://github.com/danieleades/sphinx-graph

it doesn't do anything with them right now, but i'm pretty happy with the approach. If there's anything useful in there that can be ported into sphinx-needs, be my guest!
